src/sentry/sentry_metrics/consumers/indexer/multiprocess.py

```diff
@@ -38,6 +40,8 @@ def __init__(
         self.__poll_start_time = time.time()
         self.__poll_duration_sum = 0.0
 
+        self.__last_offsets: MutableMapping[Partition, int] = defaultdict(lambda: -1)
```

Comment:
nit: This default is pretty unintuitive to me, why does it need to be a defaultdict with a default of -1? Can't a missing key just mean no last offset?

Reply:
why do you need to do all the last offset accounting anyway? aren't they ordered now so we don't need to worry about that?